libraries/chain/host_context.cpp

```diff
@@ -34,12 +38,22 @@ host_context::~host_context() = default;
 
 // called from apply_context or sync_call_context
 uint32_t host_context::execute_sync_call(name call_receiver, uint64_t flags, std::span<const char> data) {
-   auto* code = control.db().find<account_object, by_name>(call_receiver);
+   // If the call is initiated from an apply_context (an action), it is the first
+   // level of a sync call by the action, set the depth to 1.
+   // Otherwise the call is from another sync call, increment the depth by 1.
+   const uint32_t depth = dynamic_cast<apply_context*>(this) ? 1 : sync_call_depth + 1;
```

Comment:
I think we should refactor to remove this `dynamic_cast`. Can't you just pass in the depth instead?

Reply:
Yes, it can be removed because `sync_call_depth` default initial value is `0`. Now the code is cleaner. Thanks.

https://github.com/AntelopeIO/spring/pull/1285/commits/c37b00d14b30a0a79fbefc9d96e5a5b8d2feff11